lib/Installation/Partitioner/ExpertPartitionerPage.pm

```diff
@@ -109,4 +115,29 @@ sub press_accept_button {
     send_key('alt-a');
 }
 
+sub open_clone_partition_dialog {
+    my ($self) = @_;
+    assert_screen(EXPERT_PARTITIONER_PAGE);
+    send_key($self->{partition_table_shortcut});
+    for (1 .. 2) { wait_screen_change { send_key "down" } }
+    save_screenshot;
+    send_key "ret";
+}
+
+sub select_all_disks_to_clone {
+    my ($self, $numdisks) = @_;
+    assert_screen(CLONE_PARTITION);
+    send_key($self->{available_target_disks_shortcurt});
+    for (1 .. $numdisks) {
+        send_key "spc";
+        wait_screen_change { send_key "down" } unless $_==$numdisks;
```

Comment:
What's about pressing space key outside of the loop once and then switch the order to avoid this ugly condition?

Reply:
Well, not a bad idea, but then we need to change this line https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/4ec120ff72973eac1986aa3e788b47f5468b91d3/tests/installation/partitioning/raid_gpt.pm#L36 to  my $numdisks = scalar(@{$test_data->{disks}}) -2; and then have to explain in comment why -2. As otherwise, we will press space one last time and un-select the last disk. WDYT is better ?

Reply:
Not sure what you mean, here I can see http://waaa-amazing.suse.cz/tests/11054#step/raid_gpt/65 which is -1 and you start your loop from `1` not `0`, so my $numdisks = scalar(@{$test_data->{disks}}) - 1; will give 3, which makes 3 iterations.

Reply:
If we enter space once before the loop, then we need to make only two iterations instead of three otherwise the last disk will be selected twice, so it will end un-selected.

Reply:
we don't really need that condition, but it allows gaining about 5 seconds as the last send_key down is superfluous (though it does not hurt)

Reply:
ah, right. We can just start loop from `2..` saying. But please add a comment. I would also suggest asserting that all disks are selected, as that many key-presses might introduce some instability in the test execution.

Reply:
Ok, well then we would need to revert the needles already merged as last_disk_selected will be come useless. and create a new one called all_disks_selected. I did this way thinking it was more scalable. I thought it is ok as we have a wait_screen_change. It also passed 150 VRs with very high load on my WS.

Reply:
Done, please check merge the needles in description under "BIS:". I had to get rid of the wait_screen_change as it was [failing](waaa-amazing.suse.cz/tests/11062#step/raid_gpt/67) so I ran all the VRs again, all pased:
http://waaa-amazing.suse.cz/tests/11065 http://waaa-amazing.suse.cz/tests/11066 http://waaa-amazing.suse.cz/tests/11067 http://waaa-amazing.suse.cz/tests/11068 http://waaa-amazing.suse.cz/tests/11069 http://waaa-amazing.suse.cz/tests/11070 http://waaa-amazing.suse.cz/tests/11071 http://waaa-amazing.suse.cz/tests/11072 http://waaa-amazing.suse.cz/tests/11073 http://waaa-amazing.suse.cz/tests/11074 http://waaa-amazing.suse.cz/tests/11075